packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -195,6 +216,12 @@ export const NotificationsContextProvider = ({
     });
   }, [isInitialized, isInitializing, user]);
 
+  useEffect(() => {
+    if (isSubscribed || !router?.query.notify) return;
+
+    onTogglePermission(NotificationPromptSource.NotificationItem);
+  }, [onTogglePermission, isSubscribed, router]);
```

Comment:
The `onTogglePermission` has to be a dependency which led to the conversion of the functions above to be used in a `useCallback`.

Reply:
Why do we need this anyway?
Not sure I follow the use for this.

Reply:
That's the handler to show the prompt of accepting permission for notifications. That's the core feature in this task.